lib/block-editor-settings.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Adds settings to the block editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds styles and __experimentalFeatures to the block editor settings.
+ *
+ * Note: The settings that are WP version specific should be handled inside the `compat` directory.
+ *
+ * @param array $settings Existing block editor settings.
+ *
+ * @return array New block editor settings.
+ */
+function gutenberg_get_block_editor_settings( $settings ) {
+	// Recreate global styles.
+	$global_styles = array();
+	$presets       = array(
+		array(
+			'css'            => 'variables',
+			'__unstableType' => 'presets',
+			'isGlobalStyles' => true,
+		),
+		array(
+			'css'            => 'presets',
+			'__unstableType' => 'presets',
+			'isGlobalStyles' => true,
+		),
+	);
+	foreach ( $presets as $preset_style ) {
+		$actual_css = gutenberg_get_global_stylesheet( array( $preset_style['css'] ) );
+		if ( '' !== $actual_css ) {
+			$preset_style['css'] = $actual_css;
+			$global_styles[]     = $preset_style;
+		}
+	}
+
+	if ( wp_theme_has_theme_json() ) {
+		$block_classes = array(
+			'css'            => 'styles',
+			'__unstableType' => 'theme',
+			'isGlobalStyles' => true,
+		);
+		$actual_css    = gutenberg_get_global_stylesheet( array( $block_classes['css'] ) );
+		if ( '' !== $actual_css ) {
+			$block_classes['css'] = $actual_css;
+			$global_styles[]      = $block_classes;
+		}
+
+		/*
+		 * Add the custom CSS as a separate stylesheet so any invalid CSS
+		 * entered by users does not break other global styles.
+		 */
+		$global_styles[] = array(
+			'css'            => gutenberg_get_global_styles_custom_css(),
+			'__unstableType' => 'user',
+			'isGlobalStyles' => true,
+		);
+	} else {
+		// If there is no `theme.json` file, ensure base layout styles are still available.
+		$block_classes = array(
+			'css'            => 'base-layout-styles',
+			'__unstableType' => 'base-layout',
+			'isGlobalStyles' => true,
+		);
+		$actual_css    = gutenberg_get_global_stylesheet( array( $block_classes['css'] ) );
+		if ( '' !== $actual_css ) {
+			$block_classes['css'] = $actual_css;
+			$global_styles[]      = $block_classes;
+		}
+	}
+
+	$settings['styles'] = array_merge( $global_styles, get_block_editor_theme_styles() );
```

Comment:
It seems this broke duotone filters which rely on updating `$settings['styles']` in the same hook. Is there a reason that this hook needs to run at `PHP_INT_MAX` priority?

Reply:
> Is there a reason that this hook needs to run at PHP_INT_MAX priority?

That's what the plugin always used for [block setting overrides](https://github.com/WordPress/gutenberg/blob/1664d5e9708d6085c003aaef2d2feb9343f0e0d6/lib/compat/wordpress-6.2/block-editor-settings.php#L31). I guess to prevent other plugins from overriding new feature settings.

I personally don't like the use of `PHP_INT_MAX`; it prevents extensibility.

Can you point me to where the Duotone filters are located? Happy to help with the fix.

Reply:
@oandregal, any thoughts on lowering the priority for the plugin overrides?

Reply:
Opened https://github.com/WordPress/gutenberg/pull/50760. I found and fixed a few other things when I was digging through all of the code that used the same hook.